### ISSUE TITLE
Investigate allocation in TagsTest on Java 19+

### ISSUE THREAD

**Issue by shakuzen** [micrometer-metrics/micrometer#3436]
See original discovery in #3431.

> `ofEmptyDoesNotAllocate()` and `andEmptyDoesNotAllocate()` in TagsTest are failing as their allocated bytes have been changed from 0 to 16 somehow. I'm not sure what makes the difference, so it might be better to investigate further with a dedicated issue if possible.

**Comment by marcingrzejszczak** [comment]
Isn't this closed via @izeye  https://github.com/micrometer-metrics/micrometer/commit/d99e5540e2f2d6e5875eadf6ae64fd44f1b5c396 ?

```java
// See https://github.com/micrometer-metrics/micrometer/issues/3436
    @Test
    @Issue("#3313")
    @DisabledIfSystemProperty(named = "java.vm.name", matches = JAVA_VM_NAME_J9_REGEX,
            disabledReason = "Sun ThreadMXBean with allocation counter not available")
    @EnabledIf("java19")
    void ofEmptyDoesNotAllocateOnJava19() {
        ofEmptyDoesNotAllocate(16);
    }
```

**Comment by izeye** [comment]
@marcingrzejszczak The commit just seems to have fixed the test on Java 19+, and @shakuzen seems to have created this issue to investigate the allocation difference further.

**Comment by marcingrzejszczak** [comment]
Ah, I misread it, I thought that this checks on any JDK higher or equal to 19. Can't we cahnge this to (or am I missing sth) ?

```java
@EnabledForJreRange(min = JAVA_19)
```

**Comment by izeye** [comment]
@marcingrzejszczak Sure, I created https://github.com/micrometer-metrics/micrometer/pull/4512 as you suggested.

I guess the reason not using `JRE.JAVA_19` was because it wasn't available at that time or at that version on the branch.

**Comment by izeye** [comment]
The tests are failing on Java 20.0.2, so I'm not sure if they have been tested on Java 20+.

See https://app.circleci.com/pipelines/github/micrometer-metrics/micrometer/5459/workflows/92926b90-6080-482b-99ff-5222df0baf84/jobs/27922

**Comment by izeye** [comment]
The difference seems to happen only on JDK 19.

I checked it as follows:

- Liberica 19: 16 bytes
- Liberica 19.0.1: 16 bytes
- Liberica 20: 0 bytes
- Liberica 21.0.1: 0 bytes

Given JDK 20+ doesn't manifest the difference, I suspect it's a bug or unintended behavior in JDK 19.

I repurposed https://github.com/micrometer-metrics/micrometer/pull/4512 to remove the JDK 19-specific tests as JDK 19 has been EOLed already.  And I think we can also close this issue as there's not much to investigate now.
